### ISSUE TITLE
Estimate().select_components removes information of valid components

### ISSUE THREAD

**Issue by HagaiHargil** [flatironinstitute/CaImAn#401]
Hi, 

For some applications I need to log the indices of the components that were eventually used in the computation, i.e. components that following estimation are considered valid. 

However, [somewhere](https://github.com/flatironinstitute/CaImAn/blob/master/demos/general/demo_pipeline.py#L240) down the pipeline you call the `Estimate().select_components` method, which includes [these](https://github.com/flatironinstitute/CaImAn/blob/master/caiman/source_extraction/cnmf/estimates.py#L640) lines, that set the `idx_components` attributes to `None`. The immediate consequence is that the rest of the pipeline "forgets" which are the true components and which are invalid. While you do filter out some variables according to these components, you don't filter out *all* attributes, for example the cell coordinates. This leaves the user without a way to examine the cell coordinates and filter them out properly.

As for the solution, I suggest simply removing these `None`s from the method implementation. If these `None`s are needed somewhere else, then you should visit each attribute of the `CNMF` object and filter out its shape to match the new shape, only including valid components, although I don't think this "solution" is wanted.

Thanks.

**Comment by agiovann** [comment]
Dear @HagaiHargil 

thank you for the feedback. Can you be more precise about the inconsistency. When I run this piece of code at the end of demo_pipeline: 


```python
for (k,v) in cnm2.estimates.__dict__.items():
    if v is not None and k is not 'optional_outputs': 
        if type(v) in [list, tuple]:
            print([k, len(v)])
        elif 'array' in str(type(v)) or 'sparse' in str(type(v)):
            print([k, v.shape])
        else:
            print([k, type(v), v])
```
I get the following output

```batch
['A', (28900, 60)]
['C', (60, 3000)]
['f', (2, 3000)]
['b', (28900, 2)]
['YrA', (60, 3000)]
['S', (60, 3000)]
['sn', (28900,)]
['g', (60,)]
['bl', (60,)]
['c1', (60,)]
['neurons_sn', (60,)]
['lam', (60,)]
['merged_ROIs', 0]
['coordinates', (60,)]
['SNR_comp', (60,)]
['r_values', (60,)]
['cnn_preds', (60,)]
['dims', 2]
['shifts', 0]
['nr', <class 'int'>, 149]
```

only nr is inconsistent (I am fixing that part). As you see everything else contains the values 60 (components after filtering). Departing from this piece of code, could you tell me where is the inconsistency?

**Comment by epnev** [comment]
@HagaiHargil This is fixed with #404 
The rejected components are being saved by default as a new Estimates object within estimates, and can be restored (merged) with the accepted components using the estimates method `restore_discarded_components`. It's now on dev but will be merged with master soon.
